src/lib.rs

```diff
@@ -563,61 +651,75 @@ impl CANFrame {
             _id |= ERR_FLAG;
         }
 
-        let mut full_data = [0; 8];
+        let mut full_data = [0; CAN_FRAME_DATA_LEN_MAX];
 
         // not cool =/
         for (n, c) in data.iter().enumerate() {
             full_data[n] = *c;
         }
 
         Ok(CANFrame {
-               _id: _id,
-               _data_len: data.len() as u8,
-               _pad: 0,
-               _res0: 0,
-               _res1: 0,
-               _data: full_data,
-           })
+            tag: frame_type,
+            s: CANFrameStruct {
+                _id: _id,
+                _data_len: data.len() as u8,
+                _pad_flags: flags,
+                _res0: 0,
+                _res1: 0,
+                _data: full_data,
+            }
+        })
+    }
+
+    pub fn is_fd(&self) -> bool {
+        if let CANFrameType::Fd = self.tag { true } else { false }
```

Comment:
Maybe a simple ```==``` without the ```if let``` might be simpler.

Reply:
Ok, I'll change it as you propose.